desc/optimize/utils.py

```diff
@@ -533,3 +534,20 @@ def f_where_x(x, xs, fs, dim=0):
     if dim == 2:
         f = np.atleast_2d(f).reshape((-1, x.size))
     return f
+
+
+@functools.partial(jit, static_argnames="lower")
+def solve_triangular_regularized(R, b, lower=False):
+    """Solve Rx=b for triangular, possibly rank deficient R.
```

Comment:
docstring for args necessary? (i.e. I don't know what `lower` is)

Reply:
It solves Rx=b for lower or upper triangular R matrix

Reply:
it's not public, so I don't think a full doc is necessary. It's basically the same as `jax.scipy.linalg.solve_triangular` but with a bit of pre-processing.